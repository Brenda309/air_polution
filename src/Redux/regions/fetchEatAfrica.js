const FETCH_EAST_AFRICA = 'air-pollution/Redux/Regions/FETCH_EAST_AFRICA';
const initialState = [];

export const getEastCountries = () => async (dispatch) => {
  const connect = await fetch('https://restcountries.com/v3.1/region/africa');
  const res = await connect.json().then((dataObject) => dataObject);
  const dataArray = res.map((item) => {
    const dataToGet = {
      country: item.name.common,
      population: item.population,
      capital: item.capital[0],
      region: item.subregion,
      flags: item.flags.png,
      latlng: item.latlng,
    };
    return dataToGet;
  });
  const Southern = dataArray.filter((item) => item.region === 'Eastern Africa');
  dispatch({
    type: FETCH_EAST_AFRICA,
    payload: Southern,
  });
};
export function reducerForEast(state = initialState, action) {
  switch (action.type) {
    case FETCH_EAST_AFRICA:
      return action.payload;
    default:
      return state;
  }
}
