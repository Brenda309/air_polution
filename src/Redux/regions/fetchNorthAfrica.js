const FETCH_NORTH_AFRICA = 'air-pollution/Redux/Regions/FETCH_NORTH_AFRICA';
const initialState = [];

export const getNorthCountries = () => async (dispatch) => {
  const connect = await fetch('https://restcountries.com/v3.1/region/africa');
  const res = await connect.json().then((dataObject) => dataObject);
  const dataArray = res.map((item) => {
    const dataToGet = {
      country: item.name.common,
      population: item.population,
      capital: item.capital[0],
      region: item.subregion,
      flags: item.flags.png,
      latlng: item.latlng,
    };
    return dataToGet;
  });
  const northern = dataArray.filter(
    (item) => item.region === 'Northern Africa',
  );
  dispatch({
    type: FETCH_NORTH_AFRICA,
    payload: northern,
  });
};
export function reducerForNorth(state = initialState, action) {
  switch (action.type) {
    case FETCH_NORTH_AFRICA:
      return action.payload;
    default:
      return state;
  }
}
