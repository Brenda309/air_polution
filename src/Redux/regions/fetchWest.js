const FETCH_WEST_AFRICA = 'air-pollution/Redux/Regions/FETCH_WEST_AFRICA';
const initialState = [];

export const getWestCountries = () => async (dispatch) => {
  const connect = await fetch('https://restcountries.com/v3.1/region/africa');
  const res = await connect.json().then((dataObject) => dataObject);
  const dataArray = res.map((item) => {
    const dataToGet = {
      country: item.name.common,
      population: item.population,
      capital: item.capital[0],
      region: item.subregion,
      flags: item.flags.png,
      latlng: item.latlng,
    };
    return dataToGet;
  });
  const western = dataArray.filter((item) => item.region === 'Western Africa');
  dispatch({
    type: FETCH_WEST_AFRICA,
    payload: western,
  });
};
export function reducerForWest(state = initialState, action) {
  switch (action.type) {
    case FETCH_WEST_AFRICA:
      return action.payload;
    default:
      return state;
  }
}
