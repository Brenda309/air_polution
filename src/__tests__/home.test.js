import { screen, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import { reducerForNorth } from '../Redux/regions/fetchNorthAfrica';
import Home from '../components/home_page/home';
import africa from '../components/home_page/Africa.png';
import store from '../Redux/configureStore';

const mockStore = () => {
  const preloadedState = {
    northAfrica: [
      {
        country: 'Egypt',
        population: 102334403,
        capital: 'Cairo',
        region: 'Northern Africa',
        flags: 'https://flagcdn.com/w320/eg.png',
        latlng: [27.0, 30.0],
      },
      {
        country: 'Tunísia',
        population: 11818618,
        capital: 'Tunis',
        region: 'Northern Africa',
        flags: 'https://flagcdn.com/w320/tn.png',
        latlng: [34.0, 9.0],
      },
      {
        country: 'Algeria',
        population: 44700000,
        capital: 'Algiers',
        region: 'Northern Africa',
        flags: 'https://flagcdn.com/w320/dz.png',
        latlng: [28.0, 3.0],
      },
    ],
  };
  return configureStore({
    preloadedState,
    reducer: {
      northAfrica: reducerForNorth,
    },
  });
};

const mockHome = () => {
  const north = [
    {
      country: 'Egypt',
      population: 102334403,
      capital: 'Cairo',
      region: 'Northern Africa',
      flags: 'https://flagcdn.com/w320/eg.png',
      latlng: [27.0, 30.0],
    },
    {
      country: 'Tunísia',
      population: 11818618,
      capital: 'Tunis',
      region: 'Northern Africa',
      flags: 'https://flagcdn.com/w320/tn.png',
      latlng: [34.0, 9.0],
    },
    {
      country: 'Algeria',
      population: 44700000,
      capital: 'Algiers',
      region: 'Northern Africa',
      flags: 'https://flagcdn.com/w320/dz.png',
      latlng: [28.0, 3.0],
    },
  ];

  const getNorthCountries = north;

  return (
    <div className="wrapper">
      <div>
        <img id="africa" src={africa} alt="Africa-continent" />
      </div>
      <div>
        <div>
          <p className="cities">Africa</p>
        </div>
        <div>
          <p className="ncities">Cities:54</p>
        </div>
      </div>
      <p id="stats">Select capital city by region</p>
      <div className="regionCont">
        <div role="button" tabIndex={0} className="northy">
          {`Northern Africa ${getNorthCountries.length} main cities`}
        </div>
      </div>
    </div>
  );
};

describe('Testing the Home component', () => {
  test('When app runs north Africa region is displayed correctly', () => {
    render(<Provider store={mockStore()}>{mockHome()}</Provider>);
    expect(screen.getByAltText('Africa-continent')).toBeTruthy();
  });

  describe('Jest Snapshot testing suite', () => {
    it('Matches DOM Snapshot', () => {
      const { container } = render(
        <MemoryRouter>
          <Provider store={store}>
            <Home />
          </Provider>
        </MemoryRouter>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
