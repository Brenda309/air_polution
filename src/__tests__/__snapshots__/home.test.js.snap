// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Home component Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div
  class="wrapper"
>
  <div>
    <img
      alt="Africa-continent"
      id="africa"
      src="Africa.png"
    />
  </div>
  <div>
    <div>
      <p
        class="cities"
      >
        Africa
      </p>
    </div>
    <div>
      <p
        class="ncities"
      >
        Cities:54
      </p>
    </div>
  </div>
  <p
    id="stats"
  >
    Select capital city by region
  </p>
  <div
    class="regionCont"
  >
    <a
      href="/north_africa"
    >
      <div
        class="northy"
        role="button"
        tabindex="0"
      >
        Northern Africa 0 main cities
      </div>
    </a>
    <a
      href="/east_africa"
    >
      <div
        class="southy"
        role="button"
        tabindex="0"
      >
        Eastern Africa 0 main cities
      </div>
    </a>
    <a
      href="/south_africa"
    >
      <div
        class="westy"
        role="button"
        tabindex="0"
      >
        Southern Africa 0 main cities
      </div>
    </a>
    <a
      href="/west_africa"
    >
      <div
        class="westy"
        role="button"
        tabindex="0"
      >
        western Africa 0 main cities
      </div>
    </a>
  </div>
</div>
`;
